core/src/edu/cornell/gdiac/temporary/entity/Note.java

```diff
@@ -180,25 +269,47 @@ public void setTail_thickness(float t){
 		tail_thickness = t;
 	}
 	/**
-	 * Draws this shell to the canvas
-	 *
-	 * There is only one drawing pass in this application, so you can draw the objects
-	 * in any order.
+	 * Draws this note to the canvas under a width and height restriction
+	 * This will draw the image in the original scale, and will scale the image down by the smallest possible factor
+	 * to meet the confinements
 	 *
 	 * @param canvas The drawing context
 	 */
 	public void draw(GameCanvas canvas, float widthConfine, float heightConfine) {
+		//Calculate a scale such that the entire sprite fits within both confines, but does not get distorted
+		float scale = Math.max(widthConfine/w, heightConfine/h);
 		if(nt == NoteType.HELD){
-			canvas.drawRect(x - tail_thickness/2, by, x + tail_thickness/2, y, Color.BLUE, true);
+			//The tail should be about half the width of the actual note assets
+			tail_thickness = widthConfine/2f;
+			//Set the animation frame properly
+			trailAnimator.setFrame((int)curTrailFrame);
+			endAnimator.setFrame((int)curEndFrame);
+			//Start at the bottom location, then draw until we reach the top
+			float cury = by;
+			//This for loop is cursed, but its really a while loop
+			for (;cury < y - trailHeight*tail_thickness/trailWidth; cury += trailHeight*tail_thickness/trailWidth){
```

Comment:
if your initialization does not happen in the for loop just make it a while loop lol cause this is indeed cursed

or just put the initialization in the for loop since I don't see you modifying it in the body anywhere